src/components/ShelterListItem/ShelterListItem.tsx

```diff
@@ -4,121 +4,144 @@ import { useNavigate } from 'react-router-dom';
+        <h3 className="text-lg font-semibold hover:cursor-pointer hover:text-slate-500">
           {data.name}
         </h3>
         {data.verified && <VerifiedBadge />}
       </div>
-      <h6 className={cn('font-semibold text-md', availabilityClassName)}>
-        {availability}
-      </h6>
-      <h6 className="text-muted-foreground text-sm md:text-lg font-medium">
-        {data.address}
-      </h6>
+      <h6 className={cn('font-semibold text-md', availabilityClassName)}>{availability}</h6>
+      <h6 className="text-sm font-medium text-muted-foreground md:text-lg">{data.address}</h6>
       {data.shelterSupplies.length > 0 && (
         <>
           <div className="flex flex-col gap-3">
             <Separator className="mt-2" />
-            <p className="text-muted-foreground text-sm md:text-lg font-medium">
+            <p className="text-sm font-medium text-muted-foreground md:text-lg">
               Necessita voluntários:
             </p>
-            <div className="flex gap-2 flex-wrap">
-              {volunteerTags.length == 0 ?
-                <p> Não informado. <i> (Pode ser adicionado ao clicar no abrigo) </i></p>
-                :
+            <div className="flex flex-wrap gap-2" onClick={(e) => e.stopPropagation()}>
+              {Array.isArray(volunteerTags) && volunteerTags.length > 0 ? (
                 volunteerTags.map((s, idx) => (
-                  <Chip
-                    className={getSupplyPriorityProps(s.priority).className}
-                    key={idx}
-                    label={s.supply.name}
-                  />
-                ))}
+                  <WithTooltip content={nameStatusPriority(s.priority) || ''} key={idx}>
+                    <Chip
+                      className={getSupplyPriorityProps(s.priority).className}
+                      label={s.supply.name}
+                    />
+                  </WithTooltip>
+                ))
+              ) : (
+                <p>
+                  Não informado. <i> (Pode ser adicionado ao clicar no abrigo) </i>
+                </p>
+              )}
             </div>
           </div>
           <div className="flex flex-col gap-3">
             <Separator className="mt-2" />
-            <p className="text-muted-foreground text-sm md:text-lg font-medium">
+            <p className="text-sm font-medium text-muted-foreground md:text-lg">
               Necessita urgente de doações de:
             </p>
-            <div className="flex gap-2 flex-wrap">
+            <div className="flex flex-wrap gap-2" onClick={(e) => e.stopPropagation()}>
               {tags.map((s, idx) => (
-                <Chip
-                  className={getSupplyPriorityProps(s.priority).className}
-                  key={idx}
-                  label={s.supply.name}
-                />
+                <WithTooltip content={nameStatusPriority(s.priority) || ''} key={idx}>
```

Comment:
Como comentado no outro comentário, sugiro ajustar para:
```suggestion
                <WithTooltip content={nameStatusPriority(s.priority) || ''} key={`tag-${idx}`}>
```